numba/tests/test_np_functions.py

```diff
@@ -966,6 +966,17 @@ def nonconst_side(a, v, side='left'):
         with self.assertTypingError():
             cfunc([1,2], 1, side='right')
 
+        # Test unordered values
+        a = np.array([0.29769574, 0.71649186, 0.20475563])
+        v = np.array(
+            [
+                [0.18847123, 0.39659508],
+                [0.56220006, 0.57428752],
+                [0.86720994, 0.44522637],
+            ]
+        )
+        check(a, v)
```

Comment:
> * some variations on "unsorted" values and the `side` kwarg.

It looks like the `check` method tries both `side` values (and perhaps the `left` one twice?).

Reply:
I'd be inclined to make sure those branches are hit with "unsorted" data too. Maybe testing with some random data as well? I'm not sure that Numba can absolutely guarantee matching NumPy internals, but if this sufficient, doesn't hit performance, and helps out in other projects, then I think it's a change worth making.

Reply:
> I'd be inclined to make sure those branches are hit with "unsorted" data too. Maybe testing with some random data as well?

No problem, I can set that up.


> I'm not sure that Numba can absolutely guarantee matching NumPy internals, but if this sufficient, doesn't hit performance, and helps out in other projects, then I think it's a change worth making.

It's also fine if you folks aren't comfortable making this change.  Again, it's just a closer copy of NumPy's exact implementation, and since the argument should be sorted for this to make sense, exact parity in the unsorted case is just a nicety.

Reply:
Thanks for adding additional tests. Think this should be ok because, as noted, it's closer but also doesn't explicitly make the guarantee.

Reply:
> > * some variations on "unsorted" values and the `side` kwarg.
> 
> It looks like the `check` method tries both `side` values (and perhaps the `left` one twice?).

I think that, It runs the left one twice because we check both the kwarg and non-kwarg variants and the default is left, so it gets checked twice.